### ISSUE TITLE
nftables

### ISSUE THREAD

**Issue by SAprelov** [infinet/xt_wgobfs#20]
It would be useful to have example for nftables as soon it's a default packet filter for OpenWRT (and not only).

**Comment by infinet** [comment]
Please see https://forum.openwrt.org/t/wireguard-obfuscation/146054/12 . It shall work with nftables, Basically with new OpenWrt, the old iptables rules still work.

**Comment by SAprelov** [comment]
Correct. But it's legacy. In next major versions iptables will be completely removed.

**Comment by lampotakl** [comment]
Seems like it'll require modification of nft sources.
https://zasdfgbnm.github.io/2017/09/07/Extending-nftables/

**Comment by q2dg** [comment]
This article is from 2017 (Nftables v0.7). I don't know if it's still rellevant or things have changed...it would be worth checking

**Comment by egc112** [comment]
I would also like to have nft tables examples

**Comment by infinet** [comment]
I could be wrong, but it looks like nftables does not support third-party kernel modules like iptables.

**Comment by eternal-sorrow** [comment]
Why close this? Was this implemented? I don't see anything like this in the code.

**Comment by eternal-sorrow** [comment]
If you don't know how this could be done, start [here](https://www.nftables.org/documentation/HOWTO/netfilter-hacking-HOWTO-4.html). I just did a quick Google search, I'm not a Linux kernel module developer in any capacity.

**Comment by infinet** [comment]
> If you don't know how this could be done, start [here](https://www.nftables.org/documentation/HOWTO/netfilter-hacking-HOWTO-4.html).

That is basically how this extension works. However nftables is entirely different. It lacks the mechanism for this extension to work.

**Comment by eternal-sorrow** [comment]
Okay, the question for you then: will this module work in any way if `CONFIG_NETFILTER_XTABLES` is disabled in kernel configuration? I'm not talking about userspace tools to control it I'm sure `iptables` won't work, but I'm talking about the module itself.

**Comment by infinet** [comment]
> Okay, the question for you then: will this module work in any way if `CONFIG_NETFILTER_XTABLES` is disabled in kernel configuration?

I didn't test, but most likely all xtables extensions, including this one, will stop working.

**Comment by eternal-sorrow** [comment]
This is a problem.
